Add yanked versions (#3736)
astronomer/docs: astro/runtime-version-lifecycle-policy.mdx

### Restricted runtime versions

> There are some restricted versions of the Astro runtime that have known bugs and can't be used even if you enabled [deprecated Astro Runtime versions](https://docs.astronomer.io/astro/upgrade-runtime#run-a-deprecated-astro-runtime-version). These invalid versions of the Astro Runtime have the status `"yanked":true` on `https://updates.astronomer.io/astronomer-runtime` and include a description about why the version is unavailable.

Reviewer: I'm a little torn on pointing folks to our json feed. That feels weird to me.

Author: I think we need a way to give users access to the description about why the version is available, at the time it was centralized in the JSON feed, but I agree it feels weird. 

If we want that information mirrored in the docs, It makes sense to me to also include the description in the version list, especially since the descriptions aren't too lengthy. @jwitz - do you have any opinions on it?

Reviewer 2: The benefit of the JSON is that it is the real source of truth about all Astro Runtime versions. Knowing how often these updates slip through the cracks when we have to manually update them in docs, I don't see the harm in giving users that source of truth

### Restricted runtime versions

> There are some restricted versions of the Astro runtime that have known bugs and can't be used even if you enabled [deprecated Astro Runtime versions](https://docs.astronomer.io/astro/upgrade-runtime#run-a-deprecated-astro-runtime-version).

Reviewer 2: ```suggestion
There are some restricted versions of the Astro runtime that have known bugs and can't be used even if you enabled [deprecated Astro Runtime versions](https://docs.astronomer.io/astro/upgrade-runtime#run-a-deprecated-astro-runtime-version).

```

> There are some restricted versions of the Astro runtime that have known bugs, and cannot be used, even if you have [Running a deprecated Astro Runtime version](https://docs.astronomer.io/astro/upgrade-runtime#run-a-deprecated-astro-runtime-version) enabled. These invalid versions of the Astro Runtime have a status `"yanked":true` if you query `https://updates.astronomer.io/astronomer-runtime` and include a description about why the version is unavailable.

Reviewer 2: ```suggestion
There are some restricted versions of the Astro runtime that have known bugs and can't be used even if you enabled [deprecated Astro Runtime versions](https://docs.astronomer.io/astro/upgrade-runtime#run-a-deprecated-astro-runtime-version). These invalid versions of the Astro Runtime have the status `"yanked":true` on `https://updates.astronomer.io/astronomer-runtime` and include a description about why the version is unavailable.
```
I'm surprised we called this title "Restricted Runtime versions", when we use the metadata "Yanked". Any reason we wouldn't keep the naming consistent in case someone tries to search "yanked" in the docs after trying to run a yanked version?

Author: I'm of mixed feelings about this. The metadata is `yanked` but functionally, what it does is restricts the runtimes available for deployments. Maybe the subheaders `yanked` versions?

> - 8.0.0 (also deprecated)

Reviewer 2: ```suggestion
- 8.0.0
```
Would rather only focus on the deprecated status in this list, or else this becomes yet another place to update whenever we deprecate a Runtime version. That's a hassle to manage.

- 9.0.0 - 9.6.0

> - 11.0.0, 11.1.0

Reviewer 2: ```suggestion
- 11.0.0 - 11.1.0
```
I know it's only two versions but this looks more consistent
